hypex/selectors/selector_primal_methods.py

```diff
@@ -0,0 +1,257 @@
+        col
+        for col in df.columns
+        if col not in (
+            *info_col_list,
+            *target_col_list,
+            *treatment_col_list,
+            *weights_col_list,
+        )
+    ]
+
+    numeric_col_list = [
+        col
+        for col in feature_col_list
+        if col not in category_col_list
+    ]
+
+    df.loc[:, category_col_list] = df.loc[:, category_col_list].astype('str')
+    pd_fillna_inplace(df, category_col_list, is_category=True)
+    pd_fillna_inplace(df, numeric_col_list, is_category=False)
+
+    df = df.loc[:, target_col_list + feature_col_list]
+    return (
+        feature_col_list,
+        target_col_list,
+        numeric_col_list,
+        category_col_list,
+        df
+    )
+
+
+def get_feature_importance_df(gb_model, feature_names, target_name):
+    feature_importance = pd.DataFrame(
+        gb_model.feature_importances_,
+        index=feature_names,
+        columns=[f'"{target_name}" importance']
+    ).sort_index()
+
+    feature_importance[f'"{target_name}" rank'] = feature_importance[f'"{target_name}" importance'] \
+        .rank(ascending=False, method='first').astype('int')
+    return feature_importance
+
+
+def concat_feature_importance_dfs(feature_importance_list):
+    result = pd.concat(feature_importance_list, axis=1)
+    rank_col_list = result.iloc[:, 1::2].columns.tolist()
+    result.insert(0, 'rank', result.loc[:, rank_col_list].min(axis=1))
+    result = result.sort_values(['rank'] + rank_col_list, ascending=True)
+    result['rank'] = result['rank'].rank(ascending=True, method='first').astype('int')
+    return result
+
+
+def pd_lgbm_feature_selector(
+        df: pd.DataFrame,
+        info_col_list=None,
+        target='target',
+        treatment_col=None,
+        weights_col_list=None,
+        category_col_list=None,
+        model=None) -> pd.DataFrame:
+    (
+        feature_col_list,
+        target_col_list,
+        numeric_col_list,
+        category_col_list,
+        df
+    ) = pd_input_preproc(
+        df,
+        info_col_list=info_col_list,
+        target=target,
+        treatment=treatment_col,
+        weights_col_list=weights_col_list,
+        category_col_list=category_col_list
+    )
+
+    feature_importance_list = []
+    for _target_col in target:
+        lgbm_selection_model = LGBMRegressor(
+            silent=False,
+            verbose=-1,
+            num_leaves=31,
+            n_estimators=50,
+            importance_type='gain',
+        )
+
+        lgbm_selection_model.fit(
+            df[feature_col_list],
+            df[_target_col],
+            categorical_feature=category_col_list,
+        )
+
+        feature_importance = get_feature_importance_df(
+            lgbm_selection_model, feature_col_list, _target_col)
+        feature_importance_list.append(feature_importance)
+
+    report_df = concat_feature_importance_dfs(feature_importance_list)
+    return report_df
+
+
+def pd_catboost_feature_selector(
```

Comment:
Maybe add **kwargs for all methods? 
Users will be able to configure the regressors in their own way

Reply:
Will implement in the next version